amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -536,11 +556,11 @@ public void onCharacteristicRead(BluetoothGatt gatt,
                                 Log.i(TAG, "Default MTU is set to: " + currentMtu.mtu);
                                 try {
                                     mMtu = Integer.parseInt(currentMtu.mtu);
+                                    mMaxPayloadLen = mMtu - 3;
```

Comment:
Should this also do a check like line 480/489?
```
mMaxPayloadLen = mMaxPayloadLen > 0 ? mMaxPayloadLen : 0;
```

Reply:
Can you do another PR to fix that?

Reply:
@massi-ang  https://github.com/aws/amazon-freertos-ble-android-sdk/pull/21